src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5806,6 +5908,18 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_RefAssignReturnOnly" xml:space="preserve">
     <value>Cannot ref-assign '{1}' to '{0}' because '{1}' can only escape the current method through a return statement.</value>
   </data>
+  <data name="WRN_RefAssignReturnOnly" xml:space="preserve">
+    <value>This ref-assigns '{1}' to '{0}' but '{1}' can only escape the current method through a return statement.</value>
+  </data>
+  <data name="WRN_RefAssignReturnOnly_Title" xml:space="preserve">
+    <value>This ref-assigns a value that can only escape the current method through a return statement.</value>
+  </data>
+  <data name="WRN_RefAssignNarrower" xml:space="preserve">
+    <value>The right-hand-side expression '{1}' has a narrower escape scope than the left-hand-side expression '{0}' in ref-assignment.</value>
```

Comment:
>The right-hand-side expression '{1}' has a narrower escape scope than the left-hand-side expression '{0}' in ref-assignment. [](http://example.com/codeflow?start=11&length=124)

Consider matching the corresponding `ERR_RefAssignNarrower` more closely. Following your previous example, perhaps: "This ref-assigns '{1}' to '{0}' but '{1}' has a narrower escape scope than '{0}'.